docs/README.md

```diff
@@ -1,6 +1,38 @@
 # auth-service/Docs
 
 Documentation for the auth service
+## Database
+
+The following Figure depicts the Entity Relationship Diagram of the auth-service's schema:
+
+<p align="center">
+  <img src="http://www.plantuml.com/plantuml/proxy?cache=no&src=https://raw.githubusercontent.com/spikerheado1234/auth-service/master/docs/ErDiagram.puml">
+</p>
+
+The following table depicts the types of each attribute in the `Consent` table, algonside their constraints:
+
+|Attribute| Represented Internal Type | Constraints|
+|:---------:|:---------------------------:|:------------:|
+|`id`|VARCHAR(36)|Unique - Primary Key|
+|`initiator_id`|VARCHAR(36)|Non-Null|
+|`participant_id`|VARCHAR(36)|Non-Null|
+|`credential_id`|TEXT|(Can be Null)|
+|`credential_type`|TEXT|(Can be Null)|
+|`credential_status`|TEXT|(Can be Null)|
+|`credential_payload`|TEXT|(Can be Null)|
+|`credential_challenge`|TEXT|(Can be Null)|
+
+The following table depicts the types of each attribute in the `Scope` table:
+
+|Attribute| Represented Internal Type | Constraints|
+|:---------:|:---------------------------:|:------------:|
+|`id`|VARCHAR(36)|Unique - Primary Key|
+|`consent_id`|VARCHAR(36)|Foreign Key referring to the `id` in the consent table|
+|`scope_account_type`|VARCHAR(36)|Non-Null|
+|`scope_account_number`|VARCHAR(36)|(Can be Null)|
+
+The reason attributes pertaining to `credentials` and `scope_account_number` can be null is because of the communication sequence established in between the `PISP` and the `authorisation-service`. Where, at the point of first contact with the authorisation-service, these fields are not present. They are available during later parts in the sequence of communication in between the two services.
```

Comment:
```suggestion
The reason attributes pertaining to `credentials` and `account_id` can be null is because of the communication sequence established in between the PISP and the auth service. At the point of first contact with the auth service, these fields are not present and instead are added later on in the sequence of communication in between the two services.
```

Reply:
Isn't the credential info populated by the Auth service itself? If my understanding of the sequence diagrams is correct, the service itself creates and populates all the credential attributes, except the public key (payload), when it receives the request from the switch. Thus, the only credential attribute that should be allowed to be NULL is the payload.

The scopes and the account numbers are received by the Auth service in the incoming request too. If there are no scopes, we don't need to add any row in the Scopes table. Hence, they can be Non-Nullable too.

Reply:
Yes, I agree. The only nullable value should theoretically be `challenge_payload`.